src/output.rs

```diff
@@ -63,7 +63,15 @@ impl OutputType {
             return Err(Error::InvalidPagerValueBat);
         }
 
-        let resolved_path = match grep_cli::resolve_binary(&pager.bin) {
+        // On Windows 'resolve_binary()' won't be able to find 'more' as it is a COM executable
+        // (i.e. its extension is 'com') and thus we need to specify its binary name explicitly
+        let bin = if pager.kind == PagerKind::More && cfg!(windows) {
+            "more.com"
+        } else {
+            &pager.bin
+        };
+
+        let resolved_path = match grep_cli::resolve_binary(&bin) {
```

Comment:
Thank you for taking the time to fix this.

I think your change to bat will be much smaller, if needed at all, if we wait for your upstream fix https://github.com/BurntSushi/ripgrep/pull/2523? If so, I think we should wait for the upstream fix.

Reply:
I don't think that fix will make it through if the intent is to emulate the behaviour of `CreateProcess()` API. Eh, DOS compatibility...

Reply:
Upstream fix has arrived.